modules/extension/app-schema/app-schema/src/main/java/org/geotools/data/complex/config/AppSchemaDataAccessConfigurator.java

```diff
@@ -329,12 +343,77 @@ private AttributeDescriptor getTargetDescriptor(TypeMapping dto, CoordinateRefer
         return targetDescriptor;
     }
 
+    private AttributeDescriptor retypeAddingDefaultGeometry(AttributeDescriptor descriptor,
+            String defaultGeomXPath) {
+        AttributeDescriptor newDescriptor = descriptor;
+        FeatureType type = (FeatureType) descriptor.getType();
+
+        GeometryDescriptor geom = getDefaultGeometryDescriptor(type, defaultGeomXPath);
+        if (geom != null) {
+            FeatureTypeFactoryImpl ftf = new ComplexFeatureTypeFactoryImpl();
+
+            String defGeomNamespace = type.getName().getNamespaceURI();
+            String defGeomLocalName = ComplexFeatureConstants.DEFAULT_GEOMETRY_LOCAL_NAME;
+            GeometryDescriptor defGeom = ftf.createGeometryDescriptor(geom.getType(),
+                    new NameImpl(defGeomNamespace, defGeomLocalName), geom.getMinOccurs(),
+                    geom.getMaxOccurs(), geom.isNillable(), geom.getDefaultValue());
+
+            Collection<PropertyDescriptor> descriptors = new HashSet<>(type.getDescriptors());
+            descriptors.add(defGeom);
+
+            FeatureType newType = ftf.createFeatureType(type.getName(), descriptors, defGeom,
+                    type.isAbstract(), type.getRestrictions(), type.getSuper(),
+                    type.getDescription());
+            newType.getUserData().putAll(type.getUserData());
+
+            newDescriptor = new AttributeDescriptorImpl(newType, descriptor.getName(),
+                    descriptor.getMinOccurs(), descriptor.getMaxOccurs(), descriptor.isNillable(),
+                    descriptor.getDefaultValue());
+            newDescriptor.getUserData().putAll(descriptor.getUserData());
+        } else {
+            throw new IllegalArgumentException(String.format(
+                    "Default geometry descriptor could not be found for type \"%s\" at x-path \"%s\"",
+                    descriptor.getName().toString(), defaultGeomXPath));
+        }
+
+        return newDescriptor;
+    }
+
+    private GeometryDescriptor getDefaultGeometryDescriptor(FeatureType featureType, String xpath) {
+        // directly instantiating the factory I need instead of scanning the registry improves perf.
+        FeaturePropertyAccessorFactory accessorFactory = new FeaturePropertyAccessorFactory();
+        Hints hints = new Hints(PropertyAccessorFactory.NAMESPACE_CONTEXT, namespaces);
+        PropertyAccessor accessor = accessorFactory.createPropertyAccessor(featureType.getClass(),
+                xpath, GeometryAttribute.class, hints);
+
+        GeometryDescriptor geom = null;
+        if (accessor != null) {
+            try {
+                geom = accessor.get(featureType, xpath, GeometryDescriptor.class);
+            } catch (Exception e) {
+                LOGGER.finest(
+                        String.format("Exception occurred retrieving geometry descriptor "
+                                + "at x-path \"%s\": %s", xpath, e.getMessage()));
+            }
+        }
+
+        return geom;
```

Comment:
What if the XPATH is valid, but not found? I see in the tests that a invalid xpath generates an exception, so believe it's covered, but... where is that validation happening? Here?

Reply:
In the property accessor, if the XPATH doesn't exists an exception will be throw:

```
accessor.get(featureType, xpath, GeometryDescriptor.class)
```